components/esp_modem/src/esp_modem_uart.cpp

```diff
@@ -91,7 +90,6 @@ class UartTerminal : public Terminal {
     static const size_t TASK_INIT = BIT0;
     static const size_t TASK_START = BIT1;
     static const size_t TASK_STOP = BIT2;
-    static const size_t TASK_PARAMS = BIT3;
```

Comment:
Could you elaborate on why it's no longer needed?

Reply:
Yes, I've introduced one new lock to DTE. Starting from 44bae24c we shifted the responsibility of thread safety from Terminal to UART. This fix actually finishes and secures the idea.

With this change we could actually get rid of the second lock (API-lock), employing some status variables, but I would leave this to a future PR...